### PR TITLE
New Branding - New Sign In Imagery

### PR DESCRIPTION
**Pull request by grgia**
### Details
Add new sign in imagery as part of new branding project

     

### Fixed Issues

$ https://github.com/Expensify/App/issues/12253

### Tests

Note for DEV:
To easily test without passwordless beta, go to `SignInPage.js` and replace all `Permissions.canUsePasswordlessLogins(this.props.betas);` with `Permissions.canUsePasswordlessLogins([]);`

The correct flows for passwordless can be found in the screenshots section. The correct flows without passwordless beta can be found in this comment: https://github.com/Expensify/App/pull/15737#issuecomment-1505266673 

Likewise, to test beta, use `Permissions.canUsePasswordlessLogins(['passwordless']);`

- [ ] Verify that no errors appear in the JS console

WEB / Desktop
- [ ] Visually QA Sizes for small, tablet, desktop, and wide screen sizes

All Platforms:
- [ ] On log in page, visual QA
- [ ] Type in email with 2FA enabled, visual QA Magic Code page on both small and wide screen
- [ ] Type in magic code, visual QA 2FA page on both small and wide screen
- [ ] Click "go back"
- [ ] Type in email that does not exist as an account
- [ ] visual QA Magic Code page on both small and wide screen

- [ ] Go to password page, (to do this I hardcoded `showPasswordForm ` as true and all others as false in signInPage.js), visual QA on both small and wide screen

- [ ] repeat steps going offline for each page and ensure only one offline indicator shows.

### QA Steps

- [ ] Verify that no errors appear in the JS console
- [ ] Same as tests

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

Web Sizes
| Small | Medium | Wide | X Wide |
|-----|-----|-----|-----|
|image|image|image|image|

Web
| Home | Magic Code | 2FA | Password |
|-----|-----|-----|-----|
|image|image|image|image|

Web Small
| Home | Magic Code | 2FA | Password |
|-----|-----|-----|-----|
|image|image|image|image|

Web New User
| Magic Code Small | Magic Code Wide | 
|-----|-----|
|image|image|

</details>

<details>
<summary>Mobile Web - Chrome</summary>

CHROME 

| Home | Magic Code | 2FA | 
|-----|-----|-----|
|image|image|image|
</details>

<details>
<summary>Mobile Web - Safari</summary>

SAFARI 

| Home | Magic Code | 2FA |
|-----|-----|-----|
|image|image|image|

</details>

<details>
<summary>Desktop</summary>

image

Desktop Offline

| Home | Magic Code | 2FA |
|-----|-----|-----|
|image|image|image|

</details>

<details>
<summary>iOS</summary>

IOS 
| Home | Magic Code | 2FA | Password |
|-----|-----|-----|-----|
|image|Screenshot 2023-04-06 at 11 27 46 AM|Screenshot 2023-04-06 at 11 27 52 AM|Screenshot 2023-04-11 at 2 21 50 PM|

IOS Offline 
| Home | Magic Code | 2FA | Password |
|-----|-----|-----|-----|
|image|image|image|image|

</details>

<details>
<summary>Android</summary>

ANDROID 
| Home | Magic Code | 2FA | Password |
|-----|-----|-----|-----|
|image|Screenshot 2023-04-11 at 2 12 08 PM|Screenshot 2023-04-11 at 2 12 08 PM|Screenshot 2023-04-11 at 2 21 28 PM|

ANDROID OFFLINE 
| Home | Magic Code | 2FA | Password |
|-----|-----|-----|-----|
|image|||Screenshot 2023-04-11 at 2 21 16 PM|

### CODE REVIEW

**Comment by grgia** [comment]
@shawnborton I have a couple of questions I didn't anticipate for the mobile screens-

1) How will we indicate staging / dev servers?
image vs image

2) How will this design translate onto the subsequent login forms? (magic link, 2FA, etc)
image

3) Will the login terms appear on this first page? Or will we continue using the pattern of only displaying it on the magic link/2FA screens?

**Comment by grgia** [comment]
Will update likely after ECX

**Comment by grgia** [comment]
DESKTOP FLOWS:

Desktop Sizes
| Small | Medium | Wide | X Wide |
|-----|-----|-----|-----|
|image|image|image|image|


Desktop
| Home | Magic Code | 2FA | Password |
|-----|-----|-----|-----|
|image|image|image|image|

Desktop Small
| Home | Magic Code | 2FA | Password |
|-----|-----|-----|-----|
|image|image|image|image|

Desktop New User
| Magic Code Small | Magic Code Wide | 
|-----|-----|
|image|image|

Desktop Offline - WILL UPDATE
| Home | Magic Code | 2FA | Password |
|-----|-----|-----|-----|
|HERE|HERE|HERE|HERE|

File: src/pages/signin/SignInPageLayout/SignInPageContent.js [hunk 1]
```diff
@@ -46,31 +53,51 @@ const SignInPageContent = props => (
...
-                    {props.shouldShowWelcomeText && (
-                    <View style={[styles.signInPageWelcomeTextContainer]}>
-                        <Text style={[styles.mv5, styles.textLabel, styles.h3]}>
-                            {props.welcomeText}
-                        </Text>
+                    <View style={[props.isSmallScreenWidth ? styles.gap8 : styles.gap15]}>
+                        <View style={[
+                            props.isSmallScreenWidth ? styles.alignItemsCenter : {},
+
+                            // ...(props.isSmallScreenWidth ? [styles.mb2] : [styles.mt6, styles.mb5]),
```
Comment (grgia) [inline src/pages/signin/SignInPageLayout/SignInPageContent.js:60]: delete

File: src/pages/signin/SignInPageLayout/index.js [hunk 2]
```diff
@@ -43,28 +53,62 @@ const SignInPageLayout = (props) => {
...
-                            <Footer />
+                            <View style={[styles.flex1]}>
+                                <View style={{
+                                    position: 'absolute', top: 0, left: 0, right: 0, bottom: 0, justifyContent: 'center', alignItems: 'center',
+                                }}
+                                >
+                                    <SignInHeroBackgroundImage
+                                        pointerEvents="none"
+                                        width={2000}
+                                        style={[]}
```
Comment (grgia) [inline src/pages/signin/SignInPageLayout/index.js:75]: add to variables, delete

File: src/pages/signin/SignInPageLayout/index.js [hunk 3]
```diff
@@ -43,28 +53,62 @@ const SignInPageLayout = (props) => {
...
+                                    />
+                                </View>
+                                <View style={[]}>
+                                    <View style={styles.signInPageGradient}>
+                                        <SignInGradient height="100%" preserveAspectRatio="none" />
+                                    </View>
+                                    {/* <View style={[styles.signInPageGradient]} /> */}
```
Comment (grgia) [inline src/pages/signin/SignInPageLayout/index.js:82]: delete

File: src/pages/signin/SignInPageLayout/index.js [hunk 4]
```diff
@@ -43,28 +53,62 @@ const SignInPageLayout = (props) => {
...
                 ) : (
                     <ScrollView
                         contentContainerStyle={scrollViewContentContainerStyles}
                         keyboardShouldPersistTaps="handled"
                     >
-                        <View style={[styles.flex1, StyleUtils.getMinimumHeight(containerHeight)]}>
+                        <View style={[styles.flex1, styles.flexColumn, StyleUtils.getMinimumHeight(Math.max(variables.signInContentMinHeight, containerHeight))]}>
+                            <SignInHeroBackgroundImageMobile
+                                pointerEvents="none"
+                                width={800}
```
Comment (grgia) [inline src/pages/signin/SignInPageLayout/index.js:104]: use variables

File: src/pages/signin/SignInHeroImage.js [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+        imageSize = {
+            height: variables.signInHeroImageTabletHeight,
+            width: variables.signInHeroImageTabletWidth,
+        };
+    } else {
+        imageSize = {
+            height: variables.signInHeroImageDesktopHeight,
+            width: variables.signInHeroImageDesktopWidth,
+        };
+    }
```
Comment (grgia) [inline src/pages/signin/SignInHeroImage.js:30]: maybe move to StyleUtils?

File: src/styles/StyleUtils.js [hunk 6]
```diff
@@ -789,6 +800,18 @@ function getReportWelcomeTopMarginStyle(isSmallScreenWidth) {
     };
 }
 
+/**
+ * Gets the correct top margin size for the chat welcome message based on screen dimensions
```
Comment (grgia) [inline src/styles/StyleUtils.js:804]: Whoops update this

**Comment by shawnborton** [comment]
Woo, looking good so far - thanks for the comprehensive screenshots!

**Comment by grgia** [comment]
@shawnborton do you prefer the offline indicator that sticks to the screen (will have to choose one background color, so may stick out against parts of the fade) or placing it below the image (which means it may have to be scrolled down to)

Video contains both:

https://user-images.githubusercontent.com/38015950/230332742-1e0e5789-fda3-4a1e-bf15-b0aa4b29b985.mov

**Comment by grgia** [comment]
IOS 
| Home | Magic Code | 2FA | Password |
|-----|-----|-----|-----|
|image|Screenshot 2023-04-06 at 11 27 46 AM|Screenshot 2023-04-06 at 11 27 52 AM|HERE|

IOS Offline 
| Home | Magic Code | 2FA | Password |
|-----|-----|-----|-----|
|image|Screenshot 2023-04-06 at 11 27 32 AM|Screenshot 2023-04-06 at 11 28 06 AM|HERE|

SAFARI 
| Home | Magic Code | 2FA | Password |
|-----|-----|-----|-----|
|image|image|image|HERE|

ANDROID - WILL UPDATE (my emulator will not connect to wifi)
| Home | Magic Code | 2FA | Password |
|-----|-----|-----|-----|
|image|HERE|HERE|HERE|

ANDROID OFFLINE 
| Home | Magic Code | 2FA | Password |
|-----|-----|-----|-----|
|image|HERE|HERE|HERE|

CHROME - WILL UPDATE (my emulator will not connect to wifi)
| Home | Magic Code | 2FA | Password |
|-----|-----|-----|-----|
|image|HERE|HERE|HERE|

File: src/pages/signin/SignInPage.js [hunk 7]
```diff
@@ -85,27 +88,44 @@ class SignInPage extends Component {
...
+                    welcomeHeader = this.props.isSmallScreenWidth ? '' : this.props.translate('welcomeText.welcomeBack');
+                    welcomeText = this.props.isSmallScreenWidth
+                        ? `${this.props.translate('welcomeText.welcomeBack')} ${this.props.translate('welcomeText.welcomeEnterMagicCode', {login: userLogin})}`
+                        : this.props.translate('welcomeText.welcomeEnterMagicCode', {login: userLogin});
+                } else {
+                    welcomeHeader = this.props.isSmallScreenWidth ? '' : this.props.translate('welcomeText.welcome');
+                    welcomeText = this.props.isSmallScreenWidth
+                        ? `${this.props.translate('welcomeText.welcome')} ${this.props.translate('welcomeText.newFaceEnterMagicCode', {login: userLogin})}`
+                        : this.props.translate('welcomeText.newFaceEnterMagicCode', {login: userLogin});
+                }
```
Comment (grgia) [inline src/pages/signin/SignInPage.js:111]: I followed the current pattern for the title/body text, but looking at this I feel like this is hard to read, I'll think about a cleaner way to write this.

Essentially,
If on landing page / login form -> Always show header
if on any subsequent form -> Only show header on Large Screens, concat header into body for Small screens

File: src/pages/signin/SignInPage.js [hunk 8]
```diff
@@ -85,27 +88,44 @@ class SignInPage extends Component {
...
+                : this.props.translate('welcomeText.enterPassword');
         } else if (!showResendValidationForm) {
-            welcomeText = this.props.translate('welcomeText.welcome');
+            welcomeHeader = this.props.isSmallScreenWidth ? this.props.translate('login.hero.header') : this.props.translate('welcomeText.getStarted');
+            welcomeText = this.props.isSmallScreenWidth ? this.props.translate('welcomeText.getStarted') : '';
         }
 
         return (
-            <SafeAreaView style={[styles.signInPage]}>
+            <ScreenWrapper style={[styles.signInPage]}>
```
Comment (grgia) [inline src/pages/signin/SignInPage.js:124]: Use ScreenWrapper for the offline indicators on mobile

**Comment by shawnborton** [comment]
Hmm in terms of offline indicator, the one below the image definitely feels nicer but my fear is that on shorter device frames, they won't be visible? What do you think?

**Comment by grgia** [comment]
@shawnborton I agree with the offline indicators looking better underneath. But I imagine with the google/apple auth it'll get even more crowded. We could shrink to fit everything into the window size... but maybe using the other one is fine since it follows the pattern throughout the app anyways?

Also just merged main and I'll see if there's still PHP errors. Other than that, is there anything else visually I should address? Working to get this merged ASAP!

**Comment by shawnborton** [comment]
Hmm yeah, and actually with a long scrolling page, we run into some weird issues with trying to fix the offline indicator to the bottom, especially with the keyboard open right?

No other visual things that I can think of, let's get this merged and then we can follow up and put the Lottie animation in there!

**Comment by grgia** [comment]
https://github.com/Expensify/App/pull/16735#issuecomment-1500520686
@shawnborton could I get the wordmark version of this?

**Comment by grgia** [comment]
@shawnborton and @thesahindia 
Okay after sitting on it, I agree- let's move the offline indicator back below the button and above the image in that case. It will stay in the same place for desktop. It will push the image down at most.

**Comment by shawnborton** [comment]
Here is the adhoc wordmark: [expensify-logo--adhoc.svg.zip](https://github.com/Expensify/App/files/11200195/expensify-logo--adhoc.svg.zip)

**Comment by shawnborton** [comment]
I agree with that latest set of screenies for the offline indicator being the cleanest, let's go with that.

**Comment by grgia** [comment]
Sweet, that should be all the necessary updates, going to update my screenshots now!

**Comment by grgia** [comment]
LOGOS TESTS - @shawnborton I'm not sure the pills are readable enough for some of these, thoughts?

| Dev |  Staging | Production | AdHoc |
|-----|-----|-----|-----|
|image|image|image|image|
|image|image|image|image|

**Comment by shawnborton** [comment]
I think it's okay, I kinda like how small and subtle it is personally :)

**Comment by grgia** [comment]
Sweet! In that case, this is ready for review @thesahindia @AndrewGable

**Comment by grgia** [comment]
Looks like that is a bug reproducible on main. @thesahindia 

To reproduce, you log into an account with 2FA with no passwordless beta:

image

**Comment by grgia** [comment]
Update, @thesahindia this is not a bug and expected behavior. If you are not on the passwordless beta this is the flow with 2FA:

Mobile:
imageimageimage

Desktop:
imageimageimage

To easily test without beta, go to `SignInPage.js` and replace all `Permissions.canUsePasswordlessLogins(this.props.betas);` with `Permissions.canUsePasswordlessLogins(['passwordless']);`

Likewise, to test beta, use `Permissions.canUsePasswordlessLogins([]);`

**Comment by grgia** [comment]
Done! @thesahindia

**Comment by grgia** [comment]
@thesahindia this should be good for final review if you're able to prioritize it!

**Comment by grgia** [comment]
I think the password spacing is NAB for this PR. And Adhoc logo addressed [here](https://github.com/Expensify/App/pull/15737#issuecomment-1503528687).

**Review by thesahindia — APPROVED** [review]
Works well!

**Comment by grgia** [comment]
@thesahindia @AndrewGable merged main, final review!

**Review by AndrewGable — APPROVED** [review]

**Comment by AndrewGable** [comment]
🤔 - The checklist test must have been running, but checklist was all filled out.

**Pull request merged**